src/main/java/org/spongepowered/api/text/title/TitleBuilder.java

```diff
@@ -24,91 +24,311 @@
+     *
+     * @return {@code true} if the title will clear
+     * @see Title#isClear()
      */
-    TitleBuilder fadeOut(int ticks);
+    public final boolean isClear() {
+        return this.clear;
+    }
 
     /**
-     * Remove the currently displayed title from the player's screen. This will
+     * Removes the currently displayed title from the player's screen. This will
      * keep the currently used display times and will only remove the title.
      *
      * @return This title builder
+     * @see Title#isClear()
+     */
+    public TitleBuilder clear() {
+        return clear(true);
+    }
+
+    /**
+     * Sets whether the the currently displayed title should be removed from the
+     * player's screen and will keep the currently used display times.
+     *
+     * @param clear Whether this title should clear
+     * @return This title builder
+     * @see Title#isClear()
+     */
+    public TitleBuilder clear(boolean clear) {
+        if (this.clear = clear) {
+            this.title = null; // No need to send title if we clear it after
+                               // that again
+        }
+        return this;
+    }
+
+    /**
+     * Returns whether this builder is currently configured to reset.
+     *
+     * @return {@code true} if the title will reset
+     * @see Title#isReset()
+     */
+    public final boolean isReset() {
+        return this.reset;
+    }
+
+    /**
+     * Removes the currently displayed title from the player's screen and set
+     * the configuration back to the default values.
+     *
+     * @return This title builder
+     * @see Title#isReset()
      */
-    TitleBuilder clear();
+    public TitleBuilder reset() {
+        return reset(true);
+    }
 
     /**
-     * Remove the currently displayed title from the player's screen and set the
-     * configuration back to the default values.
+     * Sets whether the currently displayed title should be removed from the
+     * player's screen and the configuration set back to the default values.
      *
+     * @param reset Whether this title should reset
      * @return This title builder
+     * @see Title#isReset()
      */
-    TitleBuilder reset();
+    public TitleBuilder reset(boolean reset) {
+        if (this.reset = reset) {
+            // No need for these if we reset it again after that
+            this.title = null;
+            this.subtitle = null;
+            this.fadeIn = null;
+            this.stay = null;
+            this.fadeOut = null;
+        }
+        return this;
+    }
 
     /**
      * Builds an immutable instance of the current configuration.
      *
      * @return An immutable {@link Title} with the currently configured settings
      */
-    Title build();
+    public Title build() {
+        // Special case if the builder is empty
+        if (this.title == null && this.subtitle == null && this.fadeIn == null && this.stay == null && this.fadeOut == null) {
+            if (this.clear) {
```

Comment:
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  adam@eq2.serveftp.net
    Domain eq2.co.uk has exceeded the max emails per hour (69/60 (114%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from github-smtp2-ext5.iad.github.net ([192.30.252.196]:52588 helo=github-smtp2a-ext-cp1-prd.iad.github.net)
	by web3.cov.yesit.co.uk with esmtps (TLSv1.2:DHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.84)
	(envelope-from <noreply@github.com>)
	id 1YNebt-0003dX-1r
	for adam@eq2.co.uk; Tue, 17 Feb 2015 09:38:29 +0000
Date: Tue, 17 Feb 2015 01:38:28 -0800
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1424165908;
	bh=T7gaF/P6aeT30GaT95mBb+X20XCkDYyz4r5rlH3FbS0=;
	h=From:Reply-To:To:Cc:In-Reply-To:References:Subject:List-ID:
	 List-Archive:List-Post:List-Unsubscribe:From;
	b=ao2WDOFrLWJ+6g5R/dpxoSMk6MGdRciVwMwIHAtGYGxzsdFQ0Im4B6YK2iTpTbqfQ
	 K6b8Hg/UcIKZLAQnzkP6VG/SbdpeBYvy53DCRzNGslKhcRors1s9lGyPk5L9g5WqK0
	 B4ld7Qj8dHMEVfrnU36fyhpQMDzIdZFNCG9BE3Kc=
From: gratimax <notifications@github.com>
Reply-To: SpongePowered/SpongeAPI <reply@reply.github.com>
To: SpongePowered/SpongeAPI <SpongeAPI@noreply.github.com>
Cc: Mumfrey <adam@eq2.co.uk>
Message-ID: <SpongePowered/SpongeAPI/pull/425/r24803189@github.com>
In-Reply-To: <SpongePowered/SpongeAPI/pull/425@github.com>
References: <SpongePowered/SpongeAPI/pull/425@github.com>
Subject: Re: [SpongeAPI] [WIP] Text API 2 (#425)
Mime-Version: 1.0
Content-Type: multipart/alternative;
 boundary="--==_mimepart_54e30c14cef82_7b4e3fc728f0529c361228";
 charset=UTF-8
Content-Transfer-Encoding: 7bit
Precedence: list
X-GitHub-Sender: gratimax
X-GitHub-Recipient: Mumfrey
X-GitHub-Reason: comment
List-ID: SpongePowered/SpongeAPI <SpongeAPI.SpongePowered.github.com>
List-Archive: https://github.com/SpongePowered/SpongeAPI
List-Post: <mailto:reply@reply.github.com>
List-Unsubscribe: <mailto:unsub+001180bfbafae5ec38464915cade59baa30432a5540e88ce92cf0000000110face1492a170ce017a7775@reply.github.com>,
 <https://github.com/notifications/unsubscribe/ABGAvxEF4oARqQMeePYqYzPOWbUgpWetks5nswOUgaJpZM4DZr_C>
X-Auto-Response-Suppress: All
X-GitHub-Recipient-Address: adam@eq2.co.uk


----==_mimepart_54e30c14cef82_7b4e3fc728f0529c361228
Content-Type: text/plain;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

>  
>      /**
>       * Builds an immutable instance of the current configuration.
>       *
>       * @return An immutable {@link Title} with the currently configured settings
>       */
> -    Title build();
> +    public Title build() {
> +        // Special case if the builder is empty
> +        if (this.title == null && this.subtitle == null && this.fadeIn == null && this.stay == null && this.fadeOut == null) {
> +            if (this.clear) {

this logic here could be explained with comments, e.g. distinction between cleared and reset titles

---
Reply to this email directly or view it on GitHub:
https://github.com/SpongePowered/SpongeAPI/pull/425/files#r24803189
----==_mimepart_54e30c14cef82_7b4e3fc728f0529c361228
Content-Type: text/html;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

<p>In <a href="https://github.com/SpongePowered/SpongeAPI/pull/425#discussion_r24803189">src/main/java/org/spongepowered/api/text/title/TitleBuilder.java</a>:</p>
<pre style='color:#555'>&gt;  
&gt;      /**
&gt;       * Builds an immutable instance of the current configuration.
&gt;       *
&gt;       * @return An immutable {@link Title} with the currently configured settings
&gt;       */
&gt; -    Title build();
&gt; +    public Title build() {
&gt; +        // Special case if the builder is empty
&gt; +        if (this.title == null &amp;&amp; this.subtitle == null &amp;&amp; this.fadeIn == null &amp;&amp; this.stay == null &amp;&amp; this.fadeOut == null) {
&gt; +            if (this.clear) {
</pre>
<p>this logic here could be explained with comments, e.g. distinction between cleared and reset titles</p>

<p style="font-size:small;-webkit-text-size-adjust:none;color:#666;">&mdash;<br>Reply to this email directly or <a href="https://github.com/SpongePowered/SpongeAPI/pull/425/files#r24803189">view it on GitHub</a>.<img alt="" height="1" src="https://github.com/notifications/beacon/ABGAv17TehbDxGJM1WY4PdHcd8MKXHfCks5nswOUgaJpZM4DZr_C.gif" width="1" /></p>
<div itemscope itemtype="http://schema.org/EmailMessage">
  <div itemprop="action" itemscope itemtype="http://schema.org/ViewAction">
    <link itemprop="url" href="https://github.com/SpongePowered/SpongeAPI/pull/425/files#r24803189"></link>
    <meta itemprop="name" content="View Pull Request"></meta>
  </div>
  <meta itemprop="description" content="View this Pull Request on GitHub"></meta>
</div>

----==_mimepart_54e30c14cef82_7b4e3fc728f0529c361228--